### PR TITLE
backend environment setup

### PR DESCRIPTION
**Pull request by xiyuanzhou**

### CODE REVIEW

**Review by TrustfulDev — APPROVED** [review]
Looks good! Thanks for getting the backend environment started. I'll get Albany to start the react environment in the client folder.

File: Server/Server/settings.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+# Build paths inside the project like this: BASE_DIR / 'subdir'.
+BASE_DIR = Path(__file__).resolve().parent.parent
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/4.1/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = "django-insecure-_w=guh&@n^%ydou=0%47^!r^n-bz91se^b9cg&52xi3wnpi!9c"
```
Comment (rhedgeco) [inline Server/Server/settings.py]: I don't know if this is the actual secret key we will be using. Looks like an auto-generated django key. this should probably be replaced and also we should not keep the secret key in the public repo.

See how you would load a secret key from an environment variable instead, or however django suggests to do this in production. It looks like there is a comment above this that goes over that. We can distribute the secret key through private channels instead.

**Review by rhedgeco — CHANGES_REQUESTED** [review]
Please add a `.gitignore` file and include the `_pycache_` folder. It currently contains compiled cpython that is machine specific, plus it can bloat the repo. Other than that, looks good! Just check up on that secret key for me

**Comment by rhedgeco** [comment]
Also we should be tracking development dependencies and using python virtual environments so that multiple developers will be using the same dependencies and environment.

If you can look into making a [`requirements.txt`](https://learnpython.com/blog/python-requirements-file/) file and testing it in a fresh [`venv`](https://docs.python.org/3/library/venv.html) that would be amazing so we don't hit problems down the road. (just be sure to add the `venv` folder to the `.gitignore` too since each person will have their own)

**Review by rhedgeco — CHANGES_REQUESTED** [review]
It looks like the `_pycache_` folder is still in the branch. You will have to delete it manually and push the delete changes. Since it was already tracked, git cant untrack it until you delete it.

`.gitignore` looks good tho and gj on the `requirements.txt`

**Review by rhedgeco — APPROVED** [review]
:+1:

**Pull request merged**